### PR TITLE
Unify working with badger, remove legacy dependencies

### PR DESCRIPTION
**Pull request by deff7**

### CODE REVIEW

**Review by fat-fellow — APPROVED** [review]

File: pkg/lib/localstore/filestore/helpers.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package filestore
+
+import (
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/storage"
```
Comment (KirillSto) [inline pkg/lib/localstore/filestore/helpers.go:4]: Some files include imports in incorrect order

**Review by KirillSto — CHANGES_REQUESTED** [review]

File: util/badgerhelper/helper.go [hunk 2]
```diff
@@ -18,3 +21,91 @@ func RetryOnConflict(proc func() error) error {
 		return err
 	}
 }
+
+func Has(txn *badger.Txn, key []byte) (bool, error) {
+	_, err := txn.Get(key)
+	if err == nil {
+		return true, nil
+	}
+	if err == badger.ErrKeyNotFound {
```
Comment (AnastasiaShemyakinskaya) [inline util/badgerhelper/helper.go:30]: May be we should use errors.Is?

Comment (AnastasiaShemyakinskaya) [reply util/badgerhelper/helper.go:30]: Or `IsNotFound` function

File: pkg/lib/localstore/stores.go [hunk 3]
```diff
@@ -405,13 +233,49 @@ func GetKeysByIndex(index Index, txn ds.Txn, val interface{}, limit int) (query.
...
+
+// EraseIndex deletes the whole index
+func EraseIndex(index Index, db *badger.DB, txn *badger.Txn) (*badger.Txn, error) {
+	indexKey := IndexBase.ChildString(index.Prefix).ChildString(index.Name)
+	keys := GetKeys(txn, indexKey.String(), 0)
+
+	for _, key := range keys {
+		key := dsCtx.NewKey(key).Bytes()
+		err := txn.Delete(key)
+		if err == badger.ErrTxnTooBig {
```
Comment (AnastasiaShemyakinskaya) [inline pkg/lib/localstore/stores.go:268]: May be also use it errors.is?

File: pkg/lib/localstore/filestore/helpers.go (hunk 1 above)
Comment (deff7) [reply pkg/lib/localstore/filestore/helpers.go]: yea, thanks, accidentally turned off GCI hook

File: pkg/lib/localstore/stores.go [hunk 4]
```diff
@@ -405,13 +233,49 @@ func GetKeysByIndex(index Index, txn ds.Txn, val interface{}, limit int) (query.
...
-		Limit:    limit,
-		KeysOnly: true,
+func GetKeys(txn *badger.Txn, prefix string, limit int) []string {
+	iter := txn.NewIterator(badger.IteratorOptions{
+		Prefix:         []byte(prefix),
+		PrefetchValues: false,
 	})
+	defer iter.Close()
+
+	var keys []string
```
[the remark this answers is not in the document]
Comment (deff7) [reply pkg/lib/localstore/stores.go]: fixed

File: pkg/lib/localstore/stores.go (hunk 3 above)
Comment (deff7) [reply pkg/lib/localstore/stores.go]: done

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Pull request closed without merging**